### ISSUE TITLE
Simulation Experiment view: incorrectly marking CellML files as modified

### ISSUE THREAD

**Issue by nickerso** [opencor/opencor#1513]
In the 2017-12-23 snapshot (Mac OS), when in the simulation experiment view for a CellML document the following steps seem to flag the model as modified and I can't switch back to the editing view without "saving" the model (with no actual changes):

1. Load a CellML model and switch to simulation view.
2. Change the "ending point" value.
3. model is flagged as modified (* on the filename tab and editing tab becomes inactive)
4. "Save" the model file (which doesn't seem to change) and can then change to editing view.

(I'm sure I've seen this with other changes in the graph styling etc and simulation settings, but this is the one the reproducibly causes this effect.)

**Comment by nickerso** [comment]
Exporting a SED-ML file also seems to result in the model file being flagged as modified...

**Comment by nickerso** [comment]
hmmmm...actually the "initial_value" for my ODEs are getting updated when I have to save in these cases...which is not so useful.

**Comment by agarny** [comment]
I have just tried your steps using the `2017-12-23` snapshot on macOS, but I am not able to reproduce the behaviour you are seeing. FWIW, I tried with the models shipped with OpenCOR, i.e. `hodgkin_huxley_squid_axon_model_1952.cellml`, `noble_model_1962.cellml` and `van_der_pol_model_1928.cellml`.

In fact, for a CellML file (as opposed to a SED-ML file or COMBINE archive) to be considered modified (when using the `Simulation Experiment` view), you need to be in development mode, i.e. have clicked on the following button:

![image](https://user-images.githubusercontent.com/602265/34713302-1ee9604c-f526-11e7-806c-5e0949b4db68.png)

So, at this stage, I would need (even) more precise steps to reproduce your issue, sorry...

**Comment by nickerso** [comment]
I think I'm losing it. 

I just repeated my steps myself and the first time it was perfectly repeatable 3 times using the example `noble_model_1962.cellml`. The first time I changed the ending point after running the "default" simulation (after doing a save as to save the model outside the OpenCOR application folder) the file was flagged as modified and saving it updated the initial values for the ODEs. I then updated the ending point a second time and the file was flagged as modified but saving it produced no changes in the file. (As far as I know I am not using development mode.)

But, I've just tried this again everything is working exactly as expected. ?!?!!

I've even tried resetting everything and its still working fine...so now I don't know what is going on :) Closing until I can find a more reproducible way to demonstrate this issue (if it even exists).

**Comment by nickerso** [comment]
While I haven't found a way that is repeatable across models/sessions - this is a really annoying bug! I've also now had it flagging models as modified when adding and removing graphs after running a simulation - which makes it a pain since I can't go back to the edit mode without saving, and I need to remember to reset the model to the initial state before saving otherwise my initial_values get overridden...

yet if I do the same thing in the same OpenCOR with a different model it all works fine?!

Reopening as it really is annoying me - but feel free to re-close if needed :)

Maybe worth noting that the changes are all things that I would expect to result in a SED-ML file needing to be saved, but these are all with CellML models and not in development mode. Interestingly, I just tried toggling development mode on/off and it clears the modified flag - so that might help me avoid having to save the CellML models...

**Comment by agarny** [comment]
Ok, I have found an easy to reproduce the problem. I am sure you are glad to know that you haven't got (completely :)) insane. Anyway, to reproduce the problem:
 - Open a CellML file (e.g. `noble_model_1962.cellml`, which is shipped with OpenCOR);
 - Switch to `Simulation Experiment` mode;
 - Reload the CellML file; and
 - Edit the `Ending point`.

After that last step, the CellML file will be considered as modified while it clearly shouldn't be. FWIW, I have identified the (sill, as always) problem, so consider it fixed.
